File: src/main/java/pers/lyks/spring/example/handler/GlobalExceptionHandler.java
package pers.lyks.spring.example.handler;

import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import pers.lyks.spring.example.bean.CommonResponse;

/**
 * @author lawyerance
 * @version 1.0 2018-08-03
 */
@ControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler(Exception.class)
    public CommonResponse handleException(Exception ex) {
        CommonResponse result = new CommonResponse();
        result.setCode(19999);
        result.setData(ex.getCause());
        return result;
    }
}

File: src/main/kotlin/pers/lyks/spring/example/handler/GlobalExceptionHandler.kt
package pers.lyks.spring.example.handler

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

/**
 * @author lawyerance
 * @version 1.0 2018-08-03
 */
@ControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(Exception::class)
    fun handleException(ex: Exception): ResponseEntity<String> {
        return ResponseEntity(ex.message, HttpStatus.BAD_REQUEST)
    }
}
